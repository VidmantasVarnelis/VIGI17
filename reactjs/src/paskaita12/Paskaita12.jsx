import React, { useState } from 'react';
import Button from './components/button/Button';

const Paskaita12 = () => {
  const [count, setCount] = useState(0);

  const handleIncrement = () => setCount((prev) => prev + 1);
  const handleDecrement = () => setCount((prev) => prev - 1);

  return (
    <div>
      <h1 data-testid='heading'>Count - {count}</h1>
      <Button title='Increment' onClick={handleIncrement} />
      <Button title='Decrement' onClick={handleDecrement} />
    </div>
  );
};

export default Paskaita12;
