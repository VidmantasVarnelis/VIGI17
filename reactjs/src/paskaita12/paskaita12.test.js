import { render, screen, fireEvent } from '@testing-library/react';
import Paskaita12 from './Paskaita12';

describe('Paskaita 12 tests', () => {
  it('Paskaita 12 renders correctly', () => {
    const { container } = render(<Paskaita12 />);
    expect(container).toMatchSnapshot();
  });

  it('Button component should increase count by 1', () => {
    render(<Paskaita12 />);
    const incrementBtn = screen.getByText(/Increment/i);
    const headingEl = screen.getByTestId('heading');
    expect(headingEl.textContent).toBe('Count - 0');
    fireEvent.click(incrementBtn);
    expect(headingEl.textContent).toBe('Count - 1');
  });

  it('Button component should decrement count by 1', () => {
    render(<Paskaita12 />);
    const decrementBtn = screen.getByText(/Decrement/i);
    const headingEl = screen.getByTestId('heading');
    expect(headingEl.textContent).toBe('Count - 0');
    fireEvent.click(decrementBtn);
    expect(headingEl.textContent).toBe('Count - -1');
  });

  it('Button component should increment count by 1 and decrement count by 1', () => {
    render(<Paskaita12 />);
    const incrementBtn = screen.getByText(/Increment/i);
    const decrementBtn = screen.getByText(/Decrement/i);
    const headingEl = screen.getByTestId('heading');
    expect(headingEl.textContent).toBe('Count - 0');
    fireEvent.click(incrementBtn);
    expect(headingEl.textContent).toBe('Count - 1');
    fireEvent.click(decrementBtn);
    expect(headingEl.textContent).toBe('Count - 0');
  });

  it('Button component should increment count by 5', () => {
    render(<Paskaita12 />);
    const incrementBtn = screen.getByText(/Increment/i);
    const headingEl = screen.getByTestId('heading');
    expect(headingEl.textContent).toBe('Count - 0');
    fireEvent.click(incrementBtn);
    fireEvent.click(incrementBtn);
    fireEvent.click(incrementBtn);
    fireEvent.click(incrementBtn);
    fireEvent.click(incrementBtn);
    expect(headingEl.textContent).toBe('Count - 5');
  });
});
