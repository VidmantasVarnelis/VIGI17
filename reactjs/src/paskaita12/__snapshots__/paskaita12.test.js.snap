// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paskaita 12 tests Paskaita 12 renders correctly 1`] = `
<div>
  <div>
    <h1
      data-testid="heading"
    >
      Count - 
      0
    </h1>
    <button>
      Increment
    </button>
    <button>
      Decrement
    </button>
  </div>
</div>
`;
